lib/ansible/modules/apt_repository.py

```diff
@@ -362,10 +360,10 @@ def save(self):
                 for n, valid, enabled, source, comment in sources:
```

Comment:
I would combine the for loop https://github.com/ansible/ansible/blob/bb30a7fa1fafc8cd4dc0ff6c509cbbe524319d98/lib/ansible/modules/apt_repository.py#L340 with https://github.com/ansible/ansible/blob/bb30a7fa1fafc8cd4dc0ff6c509cbbe524319d98/lib/ansible/modules/apt_repository.py#L375 into a separate function and do the change on top of it.